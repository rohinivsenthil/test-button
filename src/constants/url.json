{
    "initialTextLink": "https://www.polvara.me/posts/how-to-test-asynchronous-methods/"
}