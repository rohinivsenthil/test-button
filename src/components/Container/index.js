import React, { Component } from 'react';
import Button from '../Button';
import TextBox from '../TextBox';
import axios from 'axios';
import url from '../../constants/url'

class Container extends Component {
    state={
        text:''
    }

    componentDidMount = async() => {
        const res = await axios.get(url.initialTextLink);
        this.setState({
            text: res.data.initialText
        })
    }

    onChange=(inputTextValue)=>{
        this.setState({
            text:inputTextValue
        })
    }

    render() {
        const {testId,testIdButton,testIdTextBox}=this.props;
        return (
            <div data-testid={testId}>
                <TextBox onChange={(this.onChange)} testId={testIdTextBox} />
                <Button text={this.state.text} buttonType="round" testId={testIdButton}/>
            </div>
        )
    }
}

export default Container
