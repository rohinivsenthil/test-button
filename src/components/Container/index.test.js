import React from 'react';
import { render, fireEvent, wait, waitForDomChange } from '@testing-library/react';
import {act} from '@testing-library/react-hooks'
import Container from '../Container';
import axios from 'axios';
import url from '../../constants'

describe('The container component',()=>{
    it('should be rendered correctly',async ()=>{
        const {asFragment}=render(<Container testId='test-cntner'/>)
        await waitForDomChange();
        expect(asFragment()).toMatchSnapshot();
    });

    it('should pass the input text entered to the button', async()=>{
        const {getByTestId}=render(<Container testId='test-cntner' testIdButton='test-btn' testIdTextBox='123'/>)
        await waitForDomChange();
        act(() => {
            fireEvent.change(getByTestId('123'),{target:{value:'Bhumika'}})
            return undefined;
        });
        expect(getByTestId('test-btn')).toHaveTextContent('Bhumika clicked 0 times.');
    })

    it('should display the content from axios.get in the button', async()=>{
        const {getByTestId}=render(<Container testId='test-cntner' testIdButton='test-btn' testIdTextBox='123'/>)
        const mockAxios = jest.spyOn(axios, 'get');
        mockAxios.mockResolvedValue({data:{initialText:'unicorn'}});
        expect(axios.get).toHaveBeenCalledWith(url);
        await wait(() => {
            expect(getByTestId('test-btn')).toHaveTextContent('unicorn')
            expect(getByTestId('123').value).toBe('unicorn')
        });

    })
    it('Should display the content from the api', async() => {
        const mockAxios = jest.spyOn(axios, 'get');
        mockAxios.mockResolvedValue({data:{initialText:'unicorn'}});

        const {getByTestId}=render(<Container testId='test-cntner' testIdButton='test-btn' testIdTextBox='123'/>)
        expect(mockAxios).toHaveBeenCalledWith(url)
        await wait(() => expect(getByTestId('123').value).toBe('unicorn'));
        await wait(() => expect(getByTestId('test-btn')).toHaveTextContent('unicorn'));
    })
})
