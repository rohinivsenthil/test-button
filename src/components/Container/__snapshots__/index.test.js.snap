// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The container component should be rendered correctly 1`] = `
<DocumentFragment>
  <div
    data-testid="test-cntner"
  >
    <div>
      <input
        type="text"
      />
    </div>
    <button>
       clicked 0 times.
    </button>
  </div>
</DocumentFragment>
`;
